### DISCUSSION TITLE
Field Depends on BelongsTo Field, BelongsTo is NULL if created via Relationship

### DISCUSSION THREAD

**Discussion by muhammadsaeedparacha** [laravel/nova-issues#5264] [Q&A] [answered]
- Laravel Version: 9.48
- Nova Version: 4.21
- PHP Version: 8.1
- Database Driver & Version:
- Operating System and Version: #.#
- Browser type and version: #.#
- Reproduction Repository: https://github.com/###/###

### Description:
When I goto Resource A
Then I create it's relationship Resource B
So that the url has `viaResource=a&viaResourceId=1`
And in Resource B, `BelongsTo` field is disabled.

Then in depends on the `$request('b') is null`
And calculations that take resource b into account here fail.

But if I am creating Resource B directly, without `viaResource=a&viaResourceId=1` in my url
Then depends on is working fine

```
class Sale extends NovaResource {
...
BelongsTo::make('Client'),
Number::make('Total'),
Number::make('Commission')->dependsOn(['client'], function (Currency $field, NovaRequest $request, FormData $formData) {
    // Here the $request->get('client') is NULL, even though it's selected
    $client = Client::find($request->get('client'));
    if ($client?->rate_type === 'percentage') {
        $field->value = $client->rate_agreed * $formData->total / 100;
    }
})
...
}
```

**Comment by crynobone** [comment, marked as answer]
This has been explained in the documentation: https://nova.laravel.com/docs/4.0/resources/fields.html#accessing-request-resource-ids
